template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/views/RdsView.java

```diff
@@ -27,43 +28,71 @@
 
     private static final String SSL_OPTIONS_WITHOUT_CERTIFICATE_FILE_PATH = "sslmode=verify-full&sslrootcert=";
 
-    private final String connectionURL;
+    private String connectionURL;
 
-    private final boolean useSsl;
+    private boolean useSsl;
 
-    private final String sslCertificateFilePath;
+    private String sslCertificateFilePath;
 
-    private final String connectionDriver;
+    private String connectionDriver;
 
-    private final String connectionUserName;
+    private String connectionUserName;
 
-    private final String connectionPassword;
+    private String connectionPassword;
 
-    private final String databaseName;
+    private String databaseName;
 
-    private final String host;
+    private String host;
 
-    private final String hostWithPortWithJdbc;
+    private String hostWithPortWithJdbc;
 
-    private final String subprotocol;
+    private String subprotocol;
 
-    private final String connectionString;
+    private String connectionString;
 
-    private final String port;
+    private String port;
 
-    private final DatabaseVendor databaseVendor;
+    private DatabaseVendor databaseVendor;
 
-    private final String withoutJDBCPrefix;
+    private String withoutJDBCPrefix;
 
-    private final RdsViewDialect rdsViewDialect;
+    private RdsViewDialect rdsViewDialect;
 
     public RdsView(RDSConfig rdsConfig) {
         this(rdsConfig, "");
     }
 
     public RdsView(RDSConfig rdsConfig, String sslCertificateFilePath) {
-        // Note: any value is valid for sslCertificateFile for sake of backward compatibility.
         this.sslCertificateFilePath = Objects.requireNonNullElse(sslCertificateFilePath, "");
+        setup(rdsConfig);
+    }
+
+    public RdsView(RdsConfigWithoutCluster rdsConfig) {
+        this(rdsConfig, "");
+    }
+
+    public RdsView(RdsConfigWithoutCluster rdsConfigWithoutCluster, String sslCertificateFilePath) {
```

Comment:
I wonder if we indeed need the `RDSConfig` constructor variants. In particular, if `RDSConfig` was to implement `RdsConfigWithoutCluster` (see my other related comments), only the `RdsConfigWithoutCluster` constructor variants would have to be kept here, and also it would not be required to have `setup()` (except if one intended to move that large logic outside the constructor, but then we would be losing the ability to make instance fields `final`).

Reply:
See my comments above